--- auto/install.orig	2010-02-01 06:01:24.000000000 -0800
+++ auto/install	2010-04-27 13:53:54.000000000 -0700
@@ -7,7 +7,7 @@
     cat << END                                                >> $NGX_MAKEFILE
 
 install_perl_modules:
-	cd $NGX_OBJS/src/http/modules/perl && make install
+	cd $NGX_OBJS/src/http/modules/perl && make pure_install
 END
 
     NGX_INSTALL_PERL_MODULES=install_perl_modules
@@ -115,8 +102,9 @@
 	test -d '\$(DESTDIR)`dirname "$NGX_HTTP_LOG_PATH"`' || \
 		mkdir -p '\$(DESTDIR)`dirname "$NGX_HTTP_LOG_PATH"`'
 
-	test -d '\$(DESTDIR)$NGX_PREFIX/html' \
-		|| cp -r html '\$(DESTDIR)$NGX_PREFIX'
+	test -d '\$(DESTDIR)$NGX_PREFIX/share/nginx/html' || \
+		mkdir -p '\$(DESTDIR)$NGX_PREFIX/share/nginx/html' && \
+		cp -r html '\$(DESTDIR)$NGX_PREFIX/share/nginx'
 END
 
 
